### PR TITLE
Fix workgroupSize

### PR DESCRIPTION
**Pull request by AMD-dwang**
When workgroupSize is defined by ExecutionModeLocalSizeId, real size
should be got after translation.

### CODE REVIEW

**Comment by AMD-dwang** [comment]
> Can you add a test for this? See the [contributing guidelines ](https://github.com/GPUOpen-Drivers/llpc/blob/dev/docs/Contributing.md#write-useful-tests) about tests.

This change don't update any instructions,  adding a test is useless. Thanks.

**Review by s-perron — CHANGES_REQUESTED** [review]
Can you add a test for a case for this?  The coverage run shows that the new code is not executed.  A new test is needed:



https://storage.googleapis.com/amdvlk-llpc-github-ci-artifacts-public/coverage_release_clang_coverage_1920555000/coverage/vulkandriver/drivers/llpc/llpc/translator/lib/SPIRV/SPIRVReader.cpp.html#L6885

**Comment by s-perron** [comment]
> > Can you add a test for this? See the [contributing guidelines ](https://github.com/GPUOpen-Drivers/llpc/blob/dev/docs/Contributing.md#write-useful-tests) about tests.
> 
> This change don't update any instructions, adding a test is useless. Thanks.

It should change the pal metadata, which can be checked in a shaderdb test.  See https://github.com/GPUOpen-Drivers/llpc/blob/f439d6b79b8cd678fc8804bd0020eaec35534e48/llpc/test/shaderdb/relocatable_shaders/PipelineVsFs_FillPsInput.pipe.  This test check the pal metadata.

**Comment by AMD-dwang** [comment]
> > > Can you add a test for this? See the [contributing guidelines ](https://github.com/GPUOpen-Drivers/llpc/blob/dev/docs/Contributing.md#write-useful-tests) about tests.
> > 
> > 
> > This change don't update any instructions, adding a test is useless. Thanks.
> 
> It should change the pal metadata, which can be checked in a shaderdb test. See https://github.com/GPUOpen-Drivers/llpc/blob/f439d6b79b8cd678fc8804bd0020eaec35534e48/llpc/test/shaderdb/relocatable_shaders/PipelineVsFs_FillPsInput.pipe. This test check the pal metadata.

I added a test, but this test don't catch this PR. The workgroupSize should come from VkPipelineShaderStageCreateInfo ::VkSpecializationInfo.

**Comment by AMD-dwang** [comment]
retest this please

**Comment by AMD-dwang** [comment]
@s-perron I have added a test. any comments?  I don't know why the checks can't be verified, but it was okay before. Thanks.

File: llpc/test/shaderdb/core/OpExecutionModeId_TestLocalSizeId.spvasm [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+OpMemberDecorate %48 0 Offset 0
+OpDecorate %48 Block
+OpDecorate %50 DescriptorSet 0
+OpDecorate %50 Binding 0
+%2 = OpTypeVoid
+%3 = OpTypeFunction %2
+%6 = OpTypeInt 32 0
+%7 = OpSpecConstant %6 2
+%8 = OpSpecConstant %6 8
+%9 = OpSpecConstant %6 8
```
Comment (s-perron) [inline llpc/test/shaderdb/core/OpExecutionModeId_TestLocalSizeId.spvasm:43]: Could you make the values for %8 and %9 different to make it easier to know that the value in the pal metadata is being taken from the correct place?

File: llpc/test/shaderdb/core/OpExecutionModeId_TestLocalSizeId.spvasm [hunk 2]
```diff
@@ -0,0 +1,99 @@
+; BEGIN_SHADERTEST
+; RUN: amdllpc -spvgen-dir=%spvgendir% -v %gfxip %s | FileCheck -check-prefix=SHADERTEST %s
+; SHADERTEST-LABEL: {{^// LLPC}} final ELF info
+; SHADERTEST: COMPUTE_NUM_THREAD_X 0x0000000000000002
```
Comment (s-perron) [inline llpc/test/shaderdb/core/OpExecutionModeId_TestLocalSizeId.spvasm:4]: Add a comment explaining where these values come from.  Something like

```
The workgroup sizes will be set to the values in the LocalSizeId instruction.  In this case, we should use the default values for the spec constants %7, %8, and %9.
```

**Review by s-perron — APPROVED** [review]
LGTM.  Just a couple suggestions.

File: llpc/test/shaderdb/core/OpExecutionModeId_TestLocalSizeId.spvasm (hunk 1 above)
Comment (AMD-dwang) [reply llpc/test/shaderdb/core/OpExecutionModeId_TestLocalSizeId.spvasm]: Updated it. Thanks.

**Review by amdrexu — APPROVED** [review]

**Pull request merged**